### PR TITLE
gut tells you if its up to date

### PR DESCRIPTION
**Pull request by astrale-sharp**
#254

The message appears at the very end of the run
every error is ignored ;)

### CODE REVIEW

**Comment by astrale-sharp** [comment]
gut1

gut2

Maybe I should jump a line before the message?

**Comment by bitwes** [comment]
Sweet!  Two small tweaks and I'll start the review

* I think we can  skip telling  the user they are  up  to date.  The less they need to read the better.
* Change the "new version" message to be:
```
GUT version <whatever> is now available.
```

**Comment by astrale-sharp** [comment]
Done :) you want me to squash them in one commit?

**Review by bitwes — CHANGES_REQUESTED** [review]
Break the code out of `_ready` into  a request method called from `_ready`.
There should at least be some tests to make sure that the handling of the response does not  cause errors.  Put these tests in `test/unit/test_utils.gd`.

There are also two bugs that need to be addressed.:

1.  If there is no internet connection then errors are printed in the output:
```
ERROR: _resolve_hostname: getaddrinfo failed! Cannot resolve hostname.
   At: drivers/unix/ip_unix.cpp:112.
[INFO]:  using [/Users/butchwesley/Library/Application Support/Godot/app_userdata/Gut] for temporary output.
Godot version:  3.2.3
GUT version:  7.1.0
ERROR: call: Error parsing JSON at line 0:
   At: modules/gdscript/gdscript_functions.cpp:1268.
```

2.   If the response does  not contain `html_url` then  GUT  hangs.  This should also be ignored.  You can  simulate this with the existing code by taking the last letter off of the url.
```
Debugger Break, Reason: 'Invalid get index 'html_url' (on base: 'Dictionary').'
*Frame 0 - res://addons/gut/utils.gd:121 in function '_http_request_completed'
Enter "help" for assistance.
```

**Comment by bitwes** [comment]
Also add a bullet point to the top of CHANGES.md

**Comment by astrale-sharp** [comment]
I've done some research but im not sure this error can be ignored `ERROR: _resolve_hostname: getaddrinfo failed! Cannot resolve hostname.
   At: drivers/unix/ip_unix.cpp:112.`

there is discussion about this [here](https://github.com/godotengine/godot/issues/22489)

File: test/unit/test_autofree.gd [hunk 1]
```diff
@@ -69,4 +69,4 @@ func test_watch_for_orphans():
 
 func test_watch_for_orphans2():
 	var n = autoqfree(Node.new())
-	assert_false(false)
\ No newline at end of file
+	assert_false(false)
```
Comment (bitwes) [inline test/unit/test_autofree.gd:72]: revert this change.

**Review by bitwes — CHANGES_REQUESTED** [review]

**Review by bitwes — APPROVED** [review]

**Pull request merged**